### PR TITLE
Remove most calls to `com`, improve `vars.Expand`

### PR DESCRIPTION
**Pull request by wxiaoguang**

### CODE REVIEW

File: modules/markup/html.go [hunk 1]
```diff
@@ -840,8 +840,8 @@ func issueIndexPatternProcessor(ctx *RenderContext, node *html.Node) {
 
 			res, err := vars.Expand(ctx.Metas["format"], ctx.Metas)
 			if err != nil {
-				log.Error(err.Error())
-				return
+				// here we could just log the error and continue the rendering
+				log.Error("unable to expand template vars for ref %s, err:%s", ref.Issue, err.Error())
```
Comment (lunny) [inline modules/markup/html.go]: But `res` here is empty?

File: modules/repository/init.go [hunk 2]
```diff
@@ -62,7 +62,8 @@ func prepareRepoCommit(ctx context.Context, repo *repo_model.Repository, tmpDir,
 	}
 	res, err := vars.Expand(string(data), match)
 	if err != nil {
-		return fmt.Errorf("expand README.md: %v", err)
+		// here we could just log the error and continue the rendering
+		log.Error("unable to expand template vars for repo README: %s, err: %s", opts.Readme, err.Error())
```
Comment (lunny) [inline modules/repository/init.go]: Is that meaningful to write an empty to the file?

File: modules/markup/html.go (hunk 1 above)
Comment (wxiaoguang) [reply modules/markup/html.go]: No, it always has content. See the Expand unit test.

File: modules/repository/init.go (hunk 2 above)
Comment (wxiaoguang) [reply modules/repository/init.go]: No, it always has content. See the Expand unit test.

**Pull request closed without merging**